### ISSUE TITLE
[BUG] Project system data flow 'Workspace update handler 42224396' closed because of an exception: Sequence contains no matching element.

### ISSUE THREAD

**Issue by Jero9999** [microsoft/vscode-dotnettools#1198]
### Describe the Issue

VS code notification. Says an unexpected error occurred in the C# dev kit. Please report it. 

Log file follows. Log file name: `ProjectSystemServerFault_c211554a-da7c-451e-b52b-b01ebe33265f.failure.txt`
```
=====================
6/14/2024 12:00:48 PM
LimitedFunctionality
System.AggregateException: Project system data flow 'Workspace update handler 42224396' closed because of an exception: Sequence contains no matching element. (Sequence contains no matching element)
---> (Inner Exception #0) StreamJsonRpc.RemoteInvocationException: Sequence contains no matching element
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass46_1`1.<<OnProjectChangedAsync>g__ApplyInBatchAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnBuildUpdateAsync(IProjectVersionedValue`1 update)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
===================
```

C# dev kit output window contents:
```
Starting Spawn .NET server...
Starting Open a solution...
Starting handling solution file "c:\<redacted>\src.api\<redacted>.sln" in Dev Kit server...
Starting Clear environment...
Using preinstalled .NET runtime at "C:\Program Files\dotnet\dotnet.exe"
.NET server started and IPC established in 2104ms
Completed Spawn .NET server (3937ms)
Completed Clear environment (3262ms)
Completed handling solution file "c:\<redacted>\src.api\<redacted>.sln" in Dev Kit server (3293ms)
Starting Restore solution...
Selected configuration: null, active configuration: Debug|Any CPU
Completed Open a solution (3321ms)
Starting NuGet restore for the solution.
Starting command: "dotnet.exe" restore C:\<redacted>\src.api\<redacted>.sln --interactive...
Completed command: "dotnet.exe" restore C:\<redacted>\src.api\<redacted>.sln --interactive (3822ms)
Completed NuGet restore.
Completed Restore solution (3825ms)
Starting NuGet restore for the solution.
Starting command: "dotnet.exe" restore C:\<redacted>\src.api\<redacted>.sln --interactive...
Failed at 'command: "dotnet.exe" restore C:\<redacted>\src.api\<redacted>.sln --interactive' (283936ms) with error: ExitCodeError: Failed with ExitCode(1)
Completed NuGet restore with error: DotnetCommandRestoreError: Failed with ExitCode(1)
```

### Steps To Reproduce

Not sure how to reproduce. Just running VSCode with my project.

### Environment Information

windows 11.

VS code version 
```
Version: 1.90.1 (user setup)
Commit: 611f9bfce64f25108829dd295f54a6894e87339d
Date: 2024-06-11T21:01:24.262Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631
```

**Comment by jchappell-cordel** [comment]
+1

I've started seeing the same issue as of today, I rolled back to the previous version (`1.6.8`) of the extension and that's resolved the problem for now.

**Comment by kvenkatrajan** [comment]
Are we able to consistently reproduce the issue? If so can you provide repro steps or a description of what flow you hit this on.

**Comment by tmeschter** [comment]
FYI @dibarbet: another report of this error.

**Comment by dibarbet** [comment]
This is not inserted into the C# extension yet, but should be in the next week or two.  Will reply here when its in a marketplace version.

**Comment by dibarbet** [comment]
this is resolved in the 2.39.28 prerelease version of the C# extension
